actors/builtin/storage_miner/storage_miner_actor.go

```diff
@@ -360,25 +426,50 @@ func (a *StorageMinerActor) DeclareTemporaryFaults(rt Runtime, sectorNumbers []a
 	})
 
 	// Request deferred Cron invocation to update temporary fault state.
-	a._rtEnrollCronEvent(rt, effectiveBeginEpoch, sectorNumbers)
-	a._rtEnrollCronEvent(rt, effectiveEndEpoch, sectorNumbers)
+	cronPayload := serde.MustSerializeParams(CronEventType_Miner_TempFault, sectorNumbers)
+	// schedule cron event to start marking temp fault at BeginEpoch
+	a._rtEnrollCronEvent(rt, effectiveBeginEpoch, cronPayload)
+	// schedule cron event to end marking temp fault at EndEpoch
+	a._rtEnrollCronEvent(rt, effectiveEndEpoch, cronPayload)
 	return &vmr.EmptyReturn{}
 }
 
 //////////
 // Cron //
 //////////
 
-func (a *StorageMinerActor) OnDeferredCronEvent(rt Runtime, sectorNumbers []abi.SectorNumber) *vmr.EmptyReturn {
+func (a *StorageMinerActor) OnDeferredCronEvent(rt Runtime, callbackPayload []byte) *vmr.EmptyReturn {
 	rt.ValidateImmediateCallerIs(builtin.StoragePowerActorAddr)
 
-	for _, sectorNumber := range sectorNumbers {
-		a._rtCheckTemporaryFaultEvents(rt, sectorNumber)
-		a._rtCheckPreCommitExpiry(rt, sectorNumber)
-		a._rtCheckSectorExpiry(rt, sectorNumber)
+	var payload CronEventPayload
+	serde.MustDeserialize(callbackPayload, payload)
```

Comment:
Deserialize only inside the branches that will use the value

Reply:
I need to deserialize to branch (the event type is inside the serialized bytes)